UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetAfterSpecifiedDateTime.cs

```diff
@@ -0,0 +1,111 @@
+            Assert.AreEqual(400, (int)apiresponse.StatusCode, $"Exchange Set for datetime is  returned {apiresponse.StatusCode}, instead of the expected 400.");
+
+            var errorMessage = await apiresponse.ReadAsTypeAsync<ErrorDescriptionResponseModel>();
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Source == "SinceDateTime"));
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Description == "Provided since date time cannot be a future date."));
+        }
+
+        
+        [TestCase("Mon, 02 Mar 2021 00:00:00 GMT", TestName = "Invalid day but valid RFC1123 Format")]
+        [TestCase("01 Mar 2021 00:00:00 GMT", TestName = "Invalid RFC format 'DD MMM YYYY HH24:MI:SS GMT'")]
+        [TestCase("01 03 2021", TestName = "Invalid RFC format 'DD MM YYYY'")]
+        public async Task WhenICallTheApiWithInValidRFC1123DateTime_ThenABadRequestStatusIsReturned(string sincedatetime)
+        {
+            var apiresponse = await ExchangesetApiClient.GetExchangeSetBasedOnDateTimeAsync(sincedatetime);
+            Assert.AreEqual(400, (int)apiresponse.StatusCode, $"Exchange Set for datetime is  returned {apiresponse.StatusCode}, instead of the expected 400.");
+
+            var errorMessage = await apiresponse.ReadAsTypeAsync<ErrorDescriptionResponseModel>();
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Source == "SinceDateTime"));
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Description == "Provided since date time is either invalid or invalid format, the valid format is 'RFC1123 format'."));
+        }
+
+
+        [TestCase("", TestName = "Provided Empty DateTime in query parameter")]
+        [TestCase(null, TestName = "Provided Null DateTime in query parameter")]
+        public async Task WhenICallTheApiWithANullDateTime_ThenABadRequestStatusIsReturned(string sincedatetime)
+        {
+            var apiresponse = await ExchangesetApiClient.GetExchangeSetBasedOnDateTimeAsync(sincedatetime);
+            Assert.AreEqual(400, (int)apiresponse.StatusCode, $"Exchange Set for datetime is  returned {apiresponse.StatusCode}, instead of the expected 400.");
+
+            var errorMessage = await apiresponse.ReadAsTypeAsync<ErrorDescriptionResponseModel>();
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Source == "SinceDateTime"));
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Description == "Query parameter 'SinceDateTime' is required."));
+        }
+
+        [TestCase("fss.ukho.gov.uk", TestName = "Callback URL without http or https")]
+        [TestCase("https:/fss.ukho.gov.uk", TestName = "Callback URL with wrong https parameter")]
+        [TestCase("http:/fss.ukho.gov.uk", TestName = "Callback URL with wrong http parameter")]
+        [TestCase("https://", TestName = "Callback URL with only https parameter")]
+        public async Task WhenICallTheApiWithInvalidCallbackURI_ThenABadRequestStatusIsReturned(string callbackurl)
```

Comment:
```suggestion
        public async Task WhenICallTheApiWithInvalidCallbackURI_ThenABadRequestResponseIsReturned(string callbackurl)
```

Reply:
Updated the Test Name as suggested. Thanks Greg.